src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,114 @@
+---
+layout: article
+title: Develop locally
+description: Learn to develop Appwrite functions locally.
+---
+
+Develop your Appwrite functions locally so you can make code changes without redeploying your function on every code change.
+
+# Setup {% #setup %}
+
+1. Install the [Docker CLI](https://www.docker.com/products/docker-desktop/)
+2. Ensure Docker is running in the background
+3. Install the [Appwrite CLI](/docs/tooling/command-line/installation#installation)
+4. [Log in](/docs/tooling/command-line/installation#get-started) to your Appwrite account using `appwrite login`
+5. [Initialize your project](/docs/tooling/command-line/installation#initialization) and create the `appwrite.json` file
+
+We use Docker to simulate the production environment precisely so that we can deploy a function locally. You can locally run functions using the `appwrite run functions` CLI command. The CLI also allows other [CLI commands](/docs/tooling/command-line/commands), and initializing a project with `appwrite.json` file is required to run commands like `appwrite run functions`.
+
+{% arrow_link href="/docs/tooling/command-line/functions" %}
+Learn more about deploying functions using CLI
+{% /arrow_link %}
+
+# Locally develop your function {% #locally-develop-your-function %}
+
+Use the `appwrite run functions` command to develop your function locally.
+
+```sh
+appwrite run functions
+? Which function would you like to develop locally? My Awesome Function ("<FUNCTION_ID>")
+
+runtime   │ entrypoint  │ path                           │ commands     
+───────────┼──────────────┼───────────────────────────────────┼─────────────
+node-16.0 │ src/main.js │ functions/"<FUNCTION_ID>"        │ npm install
+
+ℹ Info: If you wish to change your local settings, update the appwrite.json file and rerun the 'appwrite run' command.
+♥ Hint: Permissions, events, CRON and timeouts dont apply when running locally.
+ℹ Info: Pulling Docker image ...
+♥ Hint: This may take a few minutes, but we only need to do this once.
+ℹ Info: Building function using Docker ...
+Preparing for build ...
+
+Building ...
+
+
+added 4 packages, and audited 5 packages in 2s
+
+
+1 package is looking for funding
+  run `npm fund` for details
+
+
+found 0 vulnerabilities
+
+Packing build ...
+
+Build finished.
+
+ℹ Info: Starting function using Docker ...
+♥ Hint: Function automatically restarts when you edit your code.
+✓ Success: Visit http://localhost:3000/ to execute your function.
+```
+
+This command helps you efficiently develop your Appwrite functions on your local machine. When developing your Appwrite function locally, it will receive headers like when you develop a function in the Console.
```

Comment:
Please also mention hot reload

Reply:
Should I explain what hot reloading is, or should I just mention it?

Reply:
No need to go in depth. Just by mentioning what happens (by default, the CLI will listen for changes to your code and reload to apply those new changes), the developer should understand what the reload feature is and what it does.